src/overlay/FlowControl.cpp

```diff
@@ -461,7 +460,9 @@ FlowControl::addMsgAndMaybeTrimQueue(std::shared_ptr<StellarMessage const> msg)
     auto& om = mOverlayMetrics;
     if (type == TRANSACTION)
     {
-        auto isOverLimit = [&](auto const& queue) {
+        // Static analysis can't correctly infer that the lock is already held
+        // while inside the lambda, so silence it
+        auto isOverLimit = [&](auto const& queue) NO_THREAD_SAFETY_ANALYSIS {
```

Comment:
Why is this even a lambda at all? Can't this just be a boolean? It looks like we call it immediately after creating it with `queue` as the argument and never call it again elsewhere.

Reply:
Good idea, I think the lambda is a remnant we forgot to clean up post-refactoring. Switched to bool.